replication/event.go

```diff
@@ -543,6 +549,195 @@ func (e *GTIDEvent) OriginalCommitTime() time.Time {
 	return microSecTimestampToTime(e.OriginalCommitTimestamp)
 }
 
+// GtidTaggedLogEvent is for a GTID event with a tag.
+// This is similar to GTIDEvent, but it has a tag and uses a different serialization format.
+type GtidTaggedLogEvent struct {
+	GTIDEvent
+	msg serialization.Message
```

Comment:
we don't need to store `msg` field in the structure? Seems it's only used in `func (e *GtidTaggedLogEvent) Decode`. A function-local `msg` should be enough